DigitalLearningSolutions.Data/Services/DiagnosticAssessmentDataService.cs

```diff
@@ -29,6 +29,60 @@ public DiagnosticAssessmentDataService(IDbConnection connection, ILogger<Diagnos
             DiagnosticAssessment? diagnosticAssessment = null;
             return connection.Query<DiagnosticAssessment, DiagnosticTutorial, DiagnosticAssessment>(
                 @"
+                    WITH NextTutorialAndSectionNumbers AS (
+                        SELECT TOP (1)
+                            MIN(NextTutorials.TutorialID) AS NextTutorialId,
+                            MIN(NextSections.SectionID) AS NextSectionId
+                        FROM Sections
+                            INNER JOIN Customisations
+                                ON Customisations.ApplicationID = Sections.ApplicationID
+                            INNER JOIN Applications
+                                ON Applications.ApplicationID = Sections.ApplicationID
+                                                    
+                            LEFT JOIN CustomisationTutorials AS NextSectionCustomisationTutorials
+                                ON NextSectionCustomisationTutorials.CustomisationID = Customisations.CustomisationID
+                            LEFT JOIN Tutorials AS NextSectionTutorials
+                                ON NextSectionTutorials.TutorialID = NextSectionCustomisationTutorials.TutorialID
+                                AND NextSectionTutorials.ArchivedDate IS NULL
+                            LEFT JOIN Sections AS NextSections
+                                ON NextSections.SectionID = NextSectionTutorials.SectionID
+                                AND NextSections.ArchivedDate IS NULL
+                                AND (
+                                    Sections.SectionNumber < NextSections.SectionNumber
+                                    OR Sections.SectionID < NextSections.SectionID
+                                )
+                                AND (
+                                    NextSectionCustomisationTutorials.Status = 1
+                                    OR NextSectionCustomisationTutorials.DiagStatus = 1
+                                    OR NextSections.PLAssessPath IS NOT NULL
```

Comment:
For it to have a diagnostic assessment it also needs a non-null diag path. For it to have a post learning assessment it also needs to be assessed 🙂

Reply:
These conditions are based on the `GetTutorialInformation` query and are only being used to filter out invalid tutorials.
The diagnostic assessment checks are are being done in the main query below, and the post learning checks will be done in the view model.

Reply:
Sorry how is it meant to filter out invalid tutorials? Post learning assessment path doesn't have anything to do with invalid tutorials as far as I can tell? Do you mean filter out invalid sections? 

You're right that it's also used in the tutorial query so @benjimarshall can maybe explain why those checks aren't needed? 🙂

Reply:
Yeah I was beginning to respond to this, and this was an issue that I recently spotted in the tutorial content service (which has been fixed on my rewrite in #179), sorry for not passing it on to you sooner. Line 56 needs to be:
`OR (NextSectionCustomisationTutorials.DiagStatus = 1 AND NextSections.DiagAssessPath IS NOT NULL`
because for a section with just a diagnostic assessment that path needs to be present, and there needs to be a `DiagStatus=1` tutorial. If one of these were not true (and there wasn't any section to be visited, nor any post learning assessment) then this section should be skipped. Does that make sense? 🙂

Reply:
Ah my bad, I was somewhat confused because I thought this was the part of the query dealing with the next tutorial. I'll make the necessary changes now.